akka-stream/src/main/scala/akka/stream/impl/io/TcpConnectionStream.scala

```diff
@@ -64,21 +63,8 @@ private[akka] abstract class TcpStreamActor(val settings: MaterializerSettings)
     def handleRead: Receive = {
       case Received(data) ⇒
         pendingElement = data
```

Comment:
no need to store the data if closed, this should be moved into the else statement below